samcli/lib/package/packageable_resources.py

```diff
@@ -197,6 +220,21 @@ def do_export(self, resource_id, resource_dict, parent_dir):
         )
         set_value_from_jmespath(resource_dict, self.PROPERTY_NAME, {self.EXPORT_PROPERTY_CODE_KEY: uploaded_url})
 
+    def delete(self, resource_id, resource_dict):
+        """
+        Delete the ECR artifact using ECR url in PROPERTY_NAME referenced by EXPORT_PROPERTY_CODE_KEY
+        """
+        if resource_dict is None:
+            return
+
+        remote_path = resource_dict[self.PROPERTY_NAME][self.EXPORT_PROPERTY_CODE_KEY]
```

Comment:
This will raise a `KeyError` if `self.PROPERTY_NAME` is not in `resource_dict`, or if `self.EXPORT_PROPERTY_CODE_KEY]` is not in `resource_dict[self.PROPERTY_NAME]`. Are we sure both dictionaries contain the specified keys?

Reply:
Thats a good point. I was thinking of replacing that with jmespath.search() which would return None if no value is found at 'self.PROPERTY_NAME.self.EXPORT_PROPERTY_CODE_KEY'

Reply:
I think .get is enough, no need to use jmespath.search() .. You can make it 
`remote_path = resource_dict.get(self.PROPERTY_NAME, {}).get(self.EXPORT_PROPERTY_CODE_KEY)`

Reply:
okay